### PR TITLE
Replace `try-except` block with `pytest.raises`

### PR DESCRIPTION
**Pull request by rdspring1**
Currently, we use a try-except block to catch and test exceptions in the python frontend. This PR replaces that check with `pytest.raises(expected_exception, *, match: str)`, which allows checking regular expressions like `unittest.TestCase.assertRaisesRegexp`.

References:
https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions
https://docs.pytest.org/en/6.2.x/reference.html#pytest.raises

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

File: python_tests/pytest_ops.py [hunk 1]
```diff
@@ -131,23 +132,17 @@ def schedule(self):
     except Exception as e:
         exception = e
 
-    assert exception is not None, "Expected an exception"
-    exception_str = "Attempting to add to a completed definition!"
-    assert exception_str in str(
-        exception
-    ), "Failed to find correct expection error message"
+    pytest.raises(TypeError, "Attempting to add to a completed definition")
```
Comment (jjsjann123) [inline python_tests/pytest_ops.py:135]: wasn't this supposed to replace the `try/except` pattern?
I thought the example in the doc you linked suggests something like:
```
with pytest.raise(TypeError, "..."):
    fd = SchedError()
    fd.execute(...)
```

Comment (jjsjann123) [reply python_tests/pytest_ops.py:135]: NVM, I saw that this pattern is used below. Then why are we calling the pytest.raises here? and why are we catching exception here, wasn't it supposed to be caught by the pytest.raise?

**Review by jjsjann123 — COMMENTED** [review]
Thanks a lot for the link to the pytest doc.

I'm a little confused since I'm seeing two different pattern used in the PR. I pointed out the part where I think is contradicting the pattern shown in the doc.

**Comment by rdspring1** [comment]
!build

File: python_tests/pytest_ops.py (hunk 1 above)
Comment (rdspring1) [reply python_tests/pytest_ops.py]: Good catch! I must have missed changing the `definition_op_in_schedule_error_test_fn`  function.

**Review by jjsjann123 — APPROVED** [review]
LGTM!

File: python_tests/pytest_ops.py [hunk 2]
```diff
@@ -110,44 +115,34 @@ def test_correctness(op: OpInfo, dtype: torch.dtype):
...
-
     class SchedError(FusionDefinition):
         def definition(self):
-            self.t0 = fd.from_pytorch(inputs[0], static_sizes=True)
-            self.t1 = fd.ops.tanh(fd.t0)
-            self.add_output(fd.t1)
+            # Create default fusion definition
+            nvf_inputs = parse_inputs_fusion_definition(self, opinfo, *sample.args)
+            result = opinfo.op(fd)(*nvf_inputs, **sample.kwargs)
+            if type(result) is tuple:
```
Comment (jacobhinkle) [inline python_tests/pytest_ops.py:123]: Nit: `isinstance(result, tuple)`.

**Review by jacobhinkle — APPROVED** [review]
LGTM

**Comment by rdspring1** [comment]
!build

**Pull request merged**